### DISCUSSION TITLE
Running Nanos Image in Firecracker - Network Configuration

### DISCUSSION THREAD

**Discussion by Roopsai507** [nanovms/ops#1554] [Q&A] [answered]
I tried running an OPS-created image in a Firecracker VM:
```
ops image create -c config.json --package eyberg/java:1.8.0_191 --imagename javaserver
```
Here, my configuration file (`config.json`) looks like this:
```
{
    "Files":["JavaServer.jar"],
    "Args":["-jar","JavaServer.jar"],
    "RunConfig": {
        "IPAddress": "11.244.15.163",
        "NetMask": "255.255.255.252",
        "Gateway": "11.244.15.164"
   }
}
```
This setup worked successfully when I launched the image in Firecracker. Is there a way to specify network details such as IP address, gateway, and netmask directly in the kernel boot-args section of the Firecracker configuration file (`vm_config.json`), which looks like this:
```
"boot-source": {
    "kernel_image_path": "/Users/bob/.ops/0.1.26/kernel.img",
    "boot_args": "console=ttyS0 reboot=k panic=1 pci=off"
  }
```

I noticed that OSv has built-in support to pass network parameters through boot-args, as shown in the example below:
```
_cmdline = '--ip=eth0,%s,255.255.255.252 --defaultgw=%s --nameserver=%s %s' % (client_ip, tap_ip, tap_ip, cmdline)

def create_instance(self, kernel_image_path, cmdline):
        boot_source = {
            'kernel_image_path': kernel_image_path,
            'boot_args': cmdline
        }
        if self.socket_less:
            self.firecracker_config['boot-source'] = boot_source
        else:
            self.make_put_call('/boot-source', boot_source)_

```
Is there a similar way to configure network settings directly in the kernel boot-args field in Firecracker, or does OPS provide support for setting network configurations for already created images?

**Comment by eyberg** [comment, marked as answer]
I didn't know you could do this - we ignore those settings today but we could probably add that relatively easily and have them override the manifest (what ops would create)

you can also use dhcp here if you don't have a strong need for static - https://github.com/nanovms/nanos/issues/1976

**Comment by francescolavra** [comment]
With https://github.com/nanovms/nanos/pull/1983, it is now possible to configure network settings directly in the kernel boot-args field in Firecracker.
Example:
```
"boot-source": {
    "kernel_image_path": "/Users/bob/.ops/0.1.26/kernel.img",
    "boot_args": "en1.ipaddr=10.3.3.6 en1.netmask=255.255.0.0 en1.gateway=10.3.0.1"
  }
```
